### PR TITLE
A set of little improvements.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

1. Bumped eslint and its plugins versions, and autofix new errors (mostly eslint disables which were no longer neccessary)
2. ~~Replace eslint with eslint-p which speed up linter a bit (at least on my machine)~~ - CI machines don't like it.
3. Fixed/worked around one problem with flacky selection _unit_ tests.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- ~~[ ] The documentation has been updated, if necessary.~~
- ~~[ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- ~~[ ] Unit tests have been written where possible.~~
- ~~[ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.~~
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: app/common/src/appConfig.js [hunk 1]
```diff
@@ -12,9 +12,9 @@ import * as url from 'node:url'
  * environment variable. Reads from `.env` if the variable is `production`, blank or absent.
  * DOES NOT override existing environment variables if the variable is absent. */
 export async function readEnvironmentFromFile() {
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+
```
Comment (somebody1234) [inline app/common/src/appConfig.js]: i think ideally these blank lines should be removed completely

**Review by somebody1234 — APPROVED** [review]
approving dashboard changes. seems like all `eslint-disable-next-line`s that have been removed are for lints that no longer apply to the given line.

File: app/gui/src/dashboard/components/styled/withFocusScope.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import * as aria from '#/components/aria'
  * used in the component.
  */
 // This is not a React component, even though it contains JSX.
```
Comment (somebody1234) [inline app/gui/src/dashboard/components/styled/withFocusScope.tsx:14]: this comment is associated with the `no-restricted-syntax` lint and should be removed if the lint no longer applies.
(there are 5 matches here, but it's safest to do a find all to find every occurrence in the codebase)

File: app/gui/src/dashboard/data/datalinkValidator.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export const AJV = new Ajv({
 AJV.addSchema(SCHEMA)
 
 // This is a function, even though it does not contain function syntax.
```
Comment (somebody1234) [inline app/gui/src/dashboard/data/datalinkValidator.ts]: comment no longer applies

File: app/gui/src/dashboard/events/AssetEventType.ts [hunk 4]
```diff
@@ -32,5 +32,5 @@ enum AssetEventType {
 }
 
 // This is REQUIRED, as `export default enum` is invalid syntax.
```
Comment (somebody1234) [inline app/gui/src/dashboard/events/AssetEventType.ts]: comment no longer applies

File: app/gui/src/dashboard/events/AssetListEventType.ts [hunk 5]
```diff
@@ -17,5 +17,5 @@ enum AssetListEventType {
 }
 
 // This is REQUIRED, as `export default enum` is invalid syntax.
```
Comment (somebody1234) [inline app/gui/src/dashboard/events/AssetListEventType.ts]: comment no longer applies

File: app/gui/e2e/dashboard/actions/index.ts [hunk 6]
```diff
@@ -820,12 +823,12 @@ export async function passAgreementsDialog({ page }: MockParams) {
 }
 
 // This is a function, even though it does not use function syntax.
```
Comment (somebody1234) [inline app/gui/e2e/dashboard/actions/index.ts]: comment no longer applies

File: app/gui/e2e/dashboard/api.ts [hunk 7]
```diff
@@ -62,14 +62,13 @@ export interface SetupAPI {
 export type MockApi = Awaited<ReturnType<typeof mockApiInternal>>
 
 // This is a function, even though it does not contain function syntax.
```
Comment (somebody1234) [inline app/gui/e2e/dashboard/api.ts]: comment no longer applies

**Review by somebody1234 — APPROVED** [review]

**Review by kazcw — APPROVED** [review]

**Comment by farmaazon** [comment]
`eslint-p` is not welcome on our runners. So I restored the old way of linting (but with version bumps)

**Pull request merged**